ui/src/components/modals/ModelListModal.js

```diff
@@ -0,0 +1,233 @@
+                    gender
+                    race
+                    chemotherapeutic_drug_list_available
+                    age_at_aquisition
+                    disease_status
+                    files {
+                      hits{
+                        edges {
+                          node {
+                            file_name
+                            file_type
+                          }
+                        }
+                      }
+                    }
+                    therapy
+                    licensing_required
+                    date_of_availability
+                    date_created
+                    date_updated
+                    clinical_diagnosis {
+                      clinical_tumor_diagnosis
+                      aquisition_site
+                      histological_type
+                      histologcal_grade
+                      clinical_stage_grouping
+                    }
+                  }
+                }
+              }
+            }
+          }`,
+      variables: {
+        filters: { op: 'in', content: { field: '_id', value: modelIds } },
+      },
+    },
+  });
+  setState({
+    models: get(data, `models.hits.edges`, []).map(edge => edge.node),
+    loading: false,
+  });
+};
+
+export default () => (
+  <ModalStateContext.Consumer>
+    {modalState => (
+      <SelectedModelsContext.Consumer>
+        {selected => (
+          <Component
+            selected={selected}
+            modalState={modalState}
+            initialState={{ models: [], loading: true }}
+            didMount={async ({ setState }) => {
+              await fetchData({ setState, modelIds: selected.state.modelIds });
+            }}
+            didUpdate={async ({ setState, props, prevProps, state }) => {
+              // Slide the content in from right to left if not already done
+              if (modalState.state.contentRef && modalState.state.contentRef.style.right !== 0) {
+                modalState.state.contentRef.style.right = 0;
+              }
+
+              // Currently we can only remove items from this component so
+              // no need to query everytime we remove items from the list
+              const newModelId = props.selected.state.modelIds;
+              if (newModelId.length < prevProps.selected.state.modelIds.length && !state.loading) {
+                setState({
+                  ...state,
+                  models: state.models.filter(model => newModelId.indexOf(model.id) !== -1),
+                });
+              }
+            }}
+          >
+            {({ state }) => {
+              const selectedCount = selected.state.modelIds.length;
+              const hasSelected = selectedCount > 0;
+              return (
+                <>
+                  <div className="model-list-drawer-header">
+                    <h2>
+                      <DoubleArrowRightIcon width={18} height={15} />
+                      My Model List{hasSelected ? (
```

Comment:
`{hasSelected && <span className="count">{selectedCount}</span>}`, shorter and avoids the `null`